src/shell/input.c

```diff
@@ -259,15 +242,11 @@ static int shell_input_kvs_init (struct shell_input *in, json_t *header)
         goto error;
     if (!(f = flux_kvs_commit (in->shell->h, NULL, 0, txn)))
         goto error;
-    if (flux_future_then (f, -1, shell_input_kvs_init_completion, in) < 0)
-        goto error;
-    if (flux_shell_add_completion_ref (in->shell, "input.kvs-init") < 0) {
-        shell_log_errno ("flux_shell_remove_completion_ref");
-        goto error;
-    }
-    /* f memory responsibility of shell_input_kvs_init_completion()
-     * callback */
-    f = NULL;
+    if (flux_future_get (f, NULL) < 0)
+        shell_die_errno (1, "failed to create input eventlog");
+    flux_future_destroy (f);
```

Comment:
There's a `flux_future_destroy()` in the cleanup path too, so this could cause a double future destroy.